src/Scanner.h

```diff
@@ -0,0 +1,55 @@
+#ifndef SCANNER_H
+#define SCANNER_H
+
+#include <iostream>
+#include <string>
+#include <list>
+#include <map>
+#include "Token.h"
+
+
+constexpr char NULLITERAL[] = " ";
+namespace loxs {
+    class Scanner {
+        private:
+            std::string source; //Code source
+            std::list<Token> tokens; //List of tokens
+            void scanToken();
+            bool end_of_file();
+            char peek();
+            void string_literal();
+            void number_literal();
+            void identify_literal();
+            bool match(char expected);
+            char next_char();
+            long unsigned current = 0; //Points the current char of lexeme
+            long unsigned start = 0;  //Points to the first char of lexem
+            long unsigned line = 1;  //Points to the line where current is
+        public:
+            explicit Scanner(std::string code) : source(code) {}
+            std::list<Token> scanTokens();
+            void addToken(TokenType type, std::any literal);
+            void addToken(TokenType type);
+            char get_curr_char() { return source.at(current++); }
+            long unsigned get_line() { return line; }
+    };
+
+    // Map for special Keywords
+    inline const std::map<std::string, TokenType> keywords {
```

Comment:
can we make this `static const` inside the function that uses it? 😄

Reply:
I didn't put it there yet, because I don't know if is going to be needed by other function in the future

Reply:
so maybe we put it there, and then if we need it somewhere else in the future, we handle that case, for now the best place seems to be static const inside the function 😄 imo